### DISCUSSION TITLE
update_mime_database failure during demo-image-sato build, kirkstone branch

### DISCUSSION THREAD

**Discussion by hutcruchi** [OE4T/meta-tegra#1325] [Q&A] [answered]
Hi,
I try to build the tegra-demo distro and the bitbake fails with the following error:
ERROR: tegra-espimage-1.0-r0 do_rootfs: The postinstall intercept hook 'update_mime_database' failed

The relevant log:
[log.do_rootfs.82576.log](https://github.com/OE4T/meta-tegra/files/12341519/log.do_rootfs.82576.log)

The error indicating message is:`
/home/bsp/Desktop/Yocto_workspace/tegra-demo-distro/build_tegra_demo/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/intercept_scripts-3b87e4107e151ab2d2fc01c84a7e6bf3ee1c81f7bfd7aaa37e6f13cc9edf776b/update_mime_database: 8: update-mime-database: not found`

I tried two things:
1. Verified shared-mime-info package is installed
2. I saw this: 
https://community.toradex.com/t/rootfs-packages-folder-does-not-exist-build-fail-imx8mp/16050

but didn't find the file: mime.bbclass

Thanks

**Comment by hutcruchi** [comment]
I tried moving the build directory from Desktop to / . I saw some users said that sometimes too long path may cause errors so I shortened it but still get the same error.

I verified shared-mime-update command is recognised in terminal (PATH variable is ok)

**Reply by hutcruchi** [reply]
I solved this issue:

1. in the file: repos/poky/scripts/postinst-intercepts/update_mime_databse 
I updated and exported the PATH variable to include /usr/bin (to find shared-mime-update)

2. in he same file I added the code to create packages folder as mentioned here:
https://community.toradex.com/t/rootfs-packages-folder-does-not-exist-build-fail-imx8mp/16050


Now I have a different issue but I'll open other discussions for that

**Reply by madisongh** [reply]
You shouldn't have had to do that, and doing so, in fact, could cause other problems later.

It sounds like some kind of environmental issue with your build host might be causing trouble.  Have you run `fsck` on the filesystem recently?  Files not being found where they are supposed to be can be a sign of filesystem corruption.

**Reply by hutcruchi** [reply]
so you want me to restore the original update_mime_database and run fsck on my host rootFS?

**Reply by madisongh** [reply]
On whichever filesystem you're using for your builds - if that's the root filesystem, then yes.  You should not have to be hacking bits of the build system like that.

**Reply by hutcruchi** [reply]
I'll do that and update. You want me to update here or in another discussion? (here I already marked my reply as an answer)

**Reply by madisongh** [reply]
Here's fine, or the other discussion that you already started.  What's more important is getting your builds working.

**Reply by hutcruchi** [reply]
1. Thank you. I'll post here.
2. You think this problem can be related to my host ubuntu version? - I use 20.04
3. After fsck, you want me to build everything from scratch? (sourcing the build env script)

**Reply by madisongh** [reply]
Ubuntu 20.04 should be fine.

Then, yes, re-run the build (you'll likely have to reboot your system as part of the fsck pass, so re-sourcing the env setup script will be needed) and see how it goes.   Assuming fsck was able to repair any issues it found with the filesystem, of course.

**Reply by hutcruchi** [reply]
@madisongh  I used an Ubuntu rescue USB, ran fsck on my partition and found no errors (fsck -f). I tried build again from scratch and got the same error: update-mime-database: not found (and the gsk-pixbuf-quey-loaders issues)   I know that from some reason the PATH variable doesn't contains the /usr/bin directory during the update_mime_database post build script execution

**Reply by madisongh** [reply]
PATH does not include `/usr/bin` **by design** so that the commands used during the build are kept under tight control.  Hacking it in could lead to other kinds of failures.

The `update-mime-database` command should be found in `recipe-sysroot-native/usr/bin` in the per-recipe working tree, hardlinked in from the sysroots-components staging area for the `shared-mime-info-native` recipe.  If the filesystem you're using on your Ubuntu setup is something other than ext4, maybe there's an issue with how it implements hard links?

**Reply by hutcruchi** [reply]
@madisongh I'm using an ext4 filesystem. Can this problem relate to wrong/problematic branch? I downloaded the long term kirkstone

**Reply by madisongh** [reply]
We regularly run builds on all of the branches we have under active maintenance, including [kirkstone](https://builder.madison.systems/#/builders/29), so if this were a problem this basic in the branch itself, it would get caught and fixed pretty quickly.

**Reply by hutcruchi** [reply]
@madisongh anything else I can do in order to help investigate this issue? (understand what exactly got wrong and why)

**Reply by dwalkes** [reply]
@hutcruchi based on the comments above I'd try building from a completely new directory, new filesystem partition, or even a completely new build system as a troubleshooting step.  If, once you prove the build works in a new environment you still need to understand what went wrong with your original build, you can use the working system to help troubleshoot what is wrong with your current environment.

**Reply by hutcruchi** [reply]
@dwalkes I confirm I tried building the distro on a virtual machine running same ubuntu (different host) and it succeeded. 
Do you have any recommendations/starting point to investigate what's wrong on my original host system?
If its faster, I can try format it and reinstall Ubuntu.
Thanks

**Reply by dwalkes** [reply]
Have you already deleted your `build/tmp` directory and tried rebuilding?  Or just deleting your entire `build` directory and rebuilding? That's probably the easiest place to start.  Otherwise or beyond this step if you really want/need to dig into the problem I'd probably start by `find build -name "update-mime-database"` to compare where this shows up between your two different systems.. or compare the the specific file referenced above at `/home/bsp/Desktop/Yocto_workspace/tegra-demo-distro/build_tegra_demo/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/intercept_scripts-3b87e4107e151ab2d2fc01c84a7e6bf3ee1c81f7bfd7aaa37e6f13cc9edf776b/update_mime_database`

**Reply by hutcruchi** [reply, marked as answer]
I can confirm that after formatting and re installing my ubuntu host the build process completed successfully. I tried it first instead of digging deeper and maybe waste more time
